src/plugins/explore/public/components/top_nav/top_nav.tsx

```diff
@@ -150,24 +150,47 @@ export const TopNav = ({ setHeaderActionMenu = () => {}, savedExplore }: TopNavP
 
   const dispatch = useDispatch();
 
-  const handleDatasetSelect = (newDataset: any) => {
+  const handleDatasetSelect = async (newDataset: any) => {
```

Comment:
do you need to memorize this? it might be used in some dependency array like https://github.com/opensearch-project/OpenSearch-Dashboards/pull/10138